### PR TITLE
[MeshMoving] Replace OpenMP loops with parallel wrappers

### PR DESCRIPTION
**Pull request by matekelemen**
Followup on #7439 

Replace parallel loops based on OpenMP with loops using Kratos' parallel wrappers in MeshMovingApplication

### CODE REVIEW

**Comment by philbucher** [comment]
thanks @matekelemen 

I am currently finishing up #7439 so that it can be merged to master.
Afterwards you can merge master into this branch to show only the actual changes

**Comment by philbucher** [comment]
now #7439 is merged, can you please update this PR accordingly?
Thx

File: applications/MeshMovingApplication/custom_utilities/explicit_fixed_mesh_ale_utilities.cpp [hunk 1]
```diff
@@ -130,24 +132,25 @@ void ExplicitFixedMeshALEUtilities::SearchStructureNodes(
...
+    } );
 }
 
 void ExplicitFixedMeshALEUtilities::ComputeExplicitMeshDisplacement(
     const VectorResultNodesContainerType &rSearchResults,
     const DistanceVectorContainerType &rSearchDistanceResults)
 {
-    #pragma omp parallel for
-    for(int i_fl = 0; i_fl < static_cast<int>(mrVirtualModelPart.NumberOfNodes()); ++i_fl){
+    IndexPartition<size_t>( static_cast<size_t>(mrVirtualModelPart.NumberOfNodes()) ).for_each(
```
Comment (philbucher) [inline applications/MeshMovingApplication/custom_utilities/explicit_fixed_mesh_ale_utilities.cpp]:
```suggestion
    IndexPartition<std::size_t>( static_cast<std::size_t>(mrVirtualModelPart.NumberOfNodes()) ).for_each(
```

File: applications/MeshMovingApplication/custom_utilities/explicit_fixed_mesh_ale_utilities.cpp [hunk 2]
```diff
@@ -130,24 +132,25 @@ void ExplicitFixedMeshALEUtilities::SearchStructureNodes(
...
+    [&]( size_t i_fl )
```
Comment (philbucher) [inline applications/MeshMovingApplication/custom_utilities/explicit_fixed_mesh_ale_utilities.cpp]:
```suggestion
    [&]( std::size_t i_fl )
```

File: applications/MeshMovingApplication/custom_utilities/fixed_mesh_ale_utilities.cpp [hunk 3]
```diff
@@ -468,19 +471,19 @@ namespace Kratos
 
     void FixedMeshALEUtilities::RevertMeshDisplacementFixity()
     {
-        #pragma omp parallel for
-        for (int i_node = 0; i_node < static_cast<int>(mrVirtualModelPart.NumberOfNodes()); ++i_node) {
-            auto it_node = mrVirtualModelPart.NodesBegin() + i_node;
-            if (it_node->IsFixed(MESH_DISPLACEMENT_X)) {
-                it_node->Free(MESH_DISPLACEMENT_X);
+        block_for_each( mrVirtualModelPart.Nodes(),
+        []( Node<3>& node )
```
Comment (philbucher) [inline applications/MeshMovingApplication/custom_utilities/fixed_mesh_ale_utilities.cpp]:
```suggestion
        []( Node<3>& r_node )
```

@matekelemen this is how we name references
Did I show you the [Style Guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide) already?

can you please do those minor changes?
Thx

Comment (matekelemen) [reply applications/MeshMovingApplication/custom_utilities/fixed_mesh_ale_utilities.cpp]: I haven't checked the style guide before, sorry. I renamed the reference arguments and fixed a small lambda capture issue, please take a look at it and let me know if I missed anything.

**Review by philbucher — APPROVED** [review]
looks goo, thx!

merging after directly speaking with @rubenzorrilla

**Comment by rubenzorrilla** [comment]
Thanks @matekelemen

**Pull request merged**